src/api.v2/helpers/worker/workSubmit/createWorkerK8s.js

```diff
@@ -43,6 +46,45 @@ const getAvailablePods = async (namespace) => {
   return pods;
 };
 
+const getDeployment = async (name, namespace) => {
+  const k8sApi = kc.makeApiClient(k8s.AppsV1Api);
+
+  // find the particular deployment
+  const { body: deployment } = await k8sApi.readNamespacedDeployment(name, namespace);
```

Comment:
In what cases can `k8sApi.readNamespacedDeployment` return an exception? I wonder if we aren't missing some exception handling here

Reply:
My guess would be if the `name` of the deployment ('worker' hardcoded) or `namespace` didn't exist or if the api didn't have permission

Reply:
For the `getAvailablePods` it only runs it once currently. The second call is directly tied to the scaling as it needs to be called again after scaling --> I think it makes sense to keep it inside the if statement